plugins/inputs/snmp/snmp.go

```diff
@@ -857,53 +878,31 @@ func snmpTableCall(oid string) (mibName string, oidNum string, oidText string, f
 	}
 
 	mibPrefix := mibName + "::"
-	oidFullName := mibPrefix + oidText
 
 	// first attempt to get the table's tags
 	tagOids := map[string]struct{}{}
 	// We have to guess that the "entry" oid is `oid+".1"`. snmptable and snmptranslate don't seem to have a way to provide the info.
-	if out, err := execCmd("snmptranslate", "-Td", oidFullName+".1"); err == nil {
-		scanner := bufio.NewScanner(bytes.NewBuffer(out))
-		for scanner.Scan() {
-			line := scanner.Text()
-
-			if !strings.HasPrefix(line, "  INDEX") {
-				continue
-			}
-
-			i := strings.Index(line, "{ ")
-			if i == -1 { // parse error
-				continue
-			}
-			line = line[i+2:]
-			i = strings.Index(line, " }")
-			if i == -1 { // parse error
-				continue
-			}
-			line = line[:i]
-			for _, col := range strings.Split(line, ", ") {
-				tagOids[mibPrefix+col] = struct{}{}
-			}
-		}
-	}
+	// mimcks grabbing INDEX {} that is returned from snmptranslate -Td MibName
+	submask := oidNum + ".1"
+	node, err := gosmi.GetNodeByOID(types.OidMustFromString(submask))
```

Comment:
`types.OidMustFromString` will panic if there is anything wrong with the `submask`. Can we really be sure this should not happen aka there is no user input involved here? I'm asking because otherwise you could easily tear-down telegraf by crafting a malicious `submask` in here...
Long story short, maybe we should use `types.OidFromString()` and handle the error gracefully?

Reply:
This comment was in the original code `// We have to guess that the "entry" oid is `oid+".1"`. snmptable and snmptranslate don't seem to have a way to provide the info.` As well as `oidFullName+".1"` so I was trying to emulate that as closely as possible.